### PR TITLE
Dp/line hash

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Update GetHotspot request to use public API:
* updated minimum SQ version to v8.6
* added extra data fields - line hash and rule description fields.

Relates to [SLVS #1905](https://github.com/SonarSource/sonarlint-visualstudio/issues/1905)

### CODE REVIEW

File: SonarQube.Client.Tests/SonarQubeService_GetHotspotAsync.cs [hunk 1]
```diff
@@ -58,7 +58,10 @@ public async Task GetHotspot_Response_From_SonarQube()
     ""key"": ""java:S4787"",
     ""name"": ""rule-name"",
     ""securityCategory"": ""others"",
-    ""vulnerabilityProbability"": ""LOW""
+    ""vulnerabilityProbability"": ""LOW"",
+    ""riskDescription"": ""risk desc"",
+    ""vulnerabilityDescription"": ""vuln desc"",
+    ""fixRecommendations"": ""fix rec""
```
Comment (duncanp-sonar) [inline SonarQube.Client.Tests/SonarQubeService_GetHotspotAsync.cs:64]: Helpfully, the API also returns the three rule help fields, so I've included them too.

File: SonarQube.Client/Api/V8_6/GetHotspotRequest.cs [hunk 2]
```diff
@@ -38,20 +36,29 @@ public class GetHotspotRequest : RequestBase<SonarQubeHotspot>, IGetHotspotReque
...
+
+            var rule = new SonarQubeHotspotRule(
+                serverResponse.Rule.Key,
+                serverResponse.Rule.Name,
+                serverResponse.Rule.SecurityCategory,
+                serverResponse.Rule.VulnerabilityProbability,
+                serverResponse.Rule.RiskDescription,
+                serverResponse.Rule.VulnerabilityDescription,
+                serverResponse.Rule.FixRecommendations
+            );
```
Comment (duncanp-sonar) [inline SonarQube.Client/Api/V8_6/GetHotspotRequest.cs:48]: I moved the rule definition data into a separate to make it easier to work with.

File: SonarQube.Client.Tests/SonarQubeService_GetHotspotAsync.cs [hunk 3]
```diff
@@ -67,6 +70,8 @@ public async Task GetHotspot_Response_From_SonarQube()
     ""endOffset"": 41
   },
   ""status"": ""TO_REVIEW"",
+  ""line"": 32,
```
Comment (rita-g-sonarsource) [inline SonarQube.Client.Tests/SonarQubeService_GetHotspotAsync.cs:73]: Is the line field needed?
The test still asserts the line from textRange -- should we take it from here instead?

**Review by rita-g-sonarsource — APPROVED** [review]
Just the comment about the line number

File: SonarQube.Client/Api/V8_6/GetHotspotRequest.cs (hunk 2 above)
Comment (rita-g-sonarsource) [reply SonarQube.Client/Api/V8_6/GetHotspotRequest.cs:48]: Good call, it started to be a lot

**Pull request merged**